web/editor_undo_bar.js

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class EditorUndoBar {
+  #boundHide = this.hide.bind(this);
+
+  #closeButton;
+
+  #container;
+
+  #controller = null;
+
+  #eventBus;
+
+  isOpen = false;
+
+  #message;
+
+  #undoButton;
+
+  constructor({ container, message, undoButton, closeButton }, eventBus) {
+    this.#container = container;
+    this.#message = message;
+    this.#undoButton = undoButton;
+    this.#closeButton = closeButton;
+    this.#eventBus = eventBus;
+
+    // Caveat: we have to pick between registering these everytime the bar is
+    // shown and not having the ability to cleanup using AbortController.
+    this.#closeButton.addEventListener("click", this.#boundHide);
+    this.#eventBus.on("print", this.#boundHide);
+    this.#eventBus.on("download", this.#boundHide);
```

Comment:
Again, let's use the local name whenever that's available.
Also, since this is internal usage please use the "private" `EventBus`-methods instead (note that all other code in the `web/` folder does this).
```suggestion
    eventBus._on("print", this.#boundHide);
    eventBus._on("download", this.#boundHide);
```